src/aap_eda/tasks/orchestrator.py

```diff
@@ -184,4 +223,113 @@ def monitor_rulebook_processes() -> None:
         else:
             id = process.event_stream_id
         job_id = _manage_process_job_id(process_parent_type, id)
-        unique_enqueue("activation", job_id, _manage, process_parent_type, id)
+        queue = _get_queue_name(process_parent_type, id)
+        unique_enqueue(queue, job_id, _MANAGE_TASK, process_parent_type, id)
+
+
+def monitor_engine_events() -> None:
+    """Monitor container engine events.
+
+    Started by the activation worker once, before it kicks on activation
+    It enqueues a monitor task for each activation when it ends
+    """
+    LOGGER.info("Monitor engine events has started")
+    container_engine = new_container_engine(1, ProcessParentType.ACTIVATION)
+    for event in container_engine.monitor_events():
+        # monitor running instances
+        LOGGER.info(f"Container engine event {event}")
+        if event.reason in ["died", "Completed", "BackoffLimitExceeded"]:
+            for process in models.RulebookProcess.objects.filter(
+                status=ActivationStatus.RUNNING, activation_pod_id=event.name
+            ):
+                try:
+                    process_parent_type = str(process.parent_type)
+                    if process_parent_type == ProcessParentType.ACTIVATION:
+                        id = process.activation_id
+                    else:
+                        id = process.event_stream_id
+                    monitor_rulebook_process(process_parent_type, id)
+                except Exception as e:
+                    LOGGER.error(f"Monitor events error {e}")
+
+
+def _get_queue_name(
+    process_parent_type: ProcessParentType, id: int
+) -> Optional[str]:
+    obj = _get_process_parent(process_parent_type, id)
+    if obj and obj.latest_instance and obj.latest_instance.node:
+        return obj.latest_instance.node.name
+
+    return "activation"
+
+
+def _get_process_parent(process_parent_type: ProcessParentType, id: int):
+    try:
+        if process_parent_type == ProcessParentType.ACTIVATION:
+            klass = Activation
+        else:
+            klass = EventStream
+        return klass.objects.get(id=id)
+    except ObjectDoesNotExist:
+        LOGGER.warning(
+            f"{process_parent_type} with {id} no longer exists, "
+            "activation manager task will not be processed",
+        )
+    return None
+
+
+def monitor_nodes() -> None:
+    """Monitor if nodes are alive.
+
+    Started by the scheduler at a regular cadence
+    """
+    LOGGER.info("Checking Worker Health")
+    activation_values = QUEUES["activation"]
+    redis = Redis(
+        host=activation_values["HOST"], port=activation_values["PORT"]
+    )
+    now = int(time.time())
+    in_use_queues = {node.name for node in models.Node.objects.all()}
```

Comment:
How we would handle obsolete nodes? this should take into account the related activations that are in "running" state?